staging/src/k8s.io/client-go/tools/portforward/portforward.go

```diff
@@ -204,9 +204,24 @@ func (pf *PortForwarder) listenOnPortAndAddress(port *ForwardedPort, protocol st
 	return nil
 }
 
+// isUnwrappedIPv6Address indicates if we have an IPv6 that is not wrapped in
+// square brackets (needed, when using with port number). If host name, IPv4
+// address, or IPv6 address that is wrapped (which cannot be parsed as IP),
+// then return false.
+func isUnwrappedIPv6Address(nameOrIP string) bool {
+	ip := net.ParseIP(nameOrIP)
+	if ip != nil && ip.To4() == nil {
+		return true // Unwrapped IPv6 address
+	}
+	return false // name, IPv4 address, or already wrapped IPv6 address
+}
+
 // getListener creates a listener on the interface targeted by the given hostname on the given port with
 // the given protocol. protocol is in net.Listen style which basically admits values like tcp, tcp4, tcp6
 func (pf *PortForwarder) getListener(protocol string, hostname string, port *ForwardedPort) (net.Listener, error) {
+	if isUnwrappedIPv6Address(hostname) {
```

Comment:
Instead of this change, just use https://golang.org/pkg/net/#JoinHostPort istead of the fmt.Sprintf.